extensions/blocks/anchor-fm/anchor-fm.php

```diff
@@ -108,24 +118,36 @@ function process_anchor_params() {
 								'title' => $track['title'],
 							),
 						);
+
+						// Add insert basic template action.
+						$data['actions'][] = array(
+							'insert-episode-template',
+							array(
+								'episodeTrack'    => $track,
+								'spotifyImageUrl' => Assets::staticize_subdomain( 'https://wordpress.com/i/spotify-badge.svg' ),
+								'spotifyShowUrl'  => $spotify_show_url,
+							),
+						);
 					}
 				}
 			}
 		}
 	}
 
-	if ( ! empty( $spotify_show_url ) ) {
-		$data['spotifyShowUrl'] = $spotify_show_url;
-		if ( get_post_meta( $post->ID, 'jetpack_anchor_spotify_show', true ) !== $spotify_show_url ) {
-			update_post_meta( $post->ID, 'jetpack_anchor_spotify_show', $spotify_show_url );
-			$data['actions'][] = array(
-				'insert-spotify-badge',
-				array(
-					'image' => Assets::staticize_subdomain( 'https://wordpress.com/i/spotify-badge.svg' ),
-					'url'   => $spotify_show_url,
-				),
-			);
-		}
+	// Add Spotify Badge template action.
+	if (
+		$insert_spotify_badge && (
+			'post-new.php' !== $GLOBALS['pagenow'] || // Delegate badge insertion to podcast template.
+			is_null( $episode_id ) // Add badge since basic template depends on episode.
```

Comment:
I think there may be cases where the episode is given and the Spotify badge should be inserted too on an existing post:
- User creates a new post (`/post-new.php`).
- User clicks on the "Create a podcast episode" link the post-publish panel.
- User is redirected to Anchor, creates a new podcast, records an episode and publishes it to Spotify.
- Anchor redirects user back to WordPress.com (`/post.php?post=...&action=edit&anchor_podcast=...&anchor_episode=...&show_spotify_url=...`).

We should insert an Spotify badge in this scenario, so we have to remove the `is_null( $episode_id )` check from here.
```suggestion
			'post-new.php' !== $GLOBALS['pagenow'] // Delegate badge insertion to podcast template.
```

Reply:
commited. Thanks!